PYME/LMVis/pipeline.py

```diff
@@ -824,7 +824,10 @@ def OpenFile(self, filename= '', ds = None, clobber_recipe=True, **kwargs):
         
         self.recipe.execute()
         self.filterKeys = {}
-        self.selectDataSource('filtered_localizations') #NB - this rebuilds the pipeline
+        if 'filtered_localizations' in self.dataSources.keys():
+            self.selectDataSource('filtered_localizations') #NB - this rebuilds the pipeline
+        else:
+            self.selectDataSource(self.dataSources.keys()[0])
```

Comment:
`self.dataSources.keys()[0]` won't work on py3 (dict_keys is not subscriptable). It's also a little fragile, as `dict.keys()` is not guaranteed to have any specific order. We know we add the data source as `FitResults` (see line 793), so this is a safe (re. py3/ordering) alternative, although this probably needs to be revisited.

```suggestion
            # TODO - replace / remove this fallback with something better. This is currently required
            # when we use/abuse the pipeline in dh5view, but that should ideally be replaced with
            # something cleaner. This (and case above) should probably also be conditional on `clobber_recipe`
            # as if opening with an existing recipe we would likely want to keep selectedDataSource constant as well.
            self.selectDataSource('FitResults')
```